### PR TITLE
remove duplicated code 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
remove duplicated code

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

handler initiated twice.

```
	cc := &CertificateController{
		kubeClient: kubeClient,
		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "certificate"),
		handler:    handler,
	}

      cc.handler = handler 
```
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/certificates/certificate_controller.go [hunk 1]
```diff
@@ -77,30 +77,31 @@ func NewCertificateController(
...
-				}
-			}
-			glog.V(4).Infof("Deleting certificate request %s", csr.Name)
-			cc.enqueueCertificateRequest(obj)
-		},
+		DeleteFunc: cc.deleteCertificateRequest,
 	})
 	cc.csrLister = csrInformer.Lister()
 	cc.csrsSynced = csrInformer.Informer().HasSynced
-	cc.handler = handler
```
Comment (deads2k) [inline pkg/controller/certificates/certificate_controller.go]: What happened to this?

File: pkg/controller/certificates/certificate_controller.go [hunk 2]
```diff
@@ -77,30 +77,31 @@ func NewCertificateController(
 			glog.V(4).Infof("Updating certificate request %s", oldCSR.Name)
 			cc.enqueueCertificateRequest(new)
 		},
-		DeleteFunc: func(obj interface{}) {
```
Comment (mikedanese) [inline pkg/controller/certificates/certificate_controller.go]: why extract this?

File: pkg/controller/certificates/certificate_controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/controller/certificates/certificate_controller.go]: upper initiated

File: pkg/controller/certificates/certificate_controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/controller/certificates/certificate_controller.go]: I think this moved out from new controller is better from coding view as all other controllers, Which makes code more readable

**Comment by hzxuzhonghu** [comment]
@deads2k please take abother look

**Pull request merged**